From 6e62a468a0c98474bad47adcb7a3a561706b4291 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 24 Aug 2022 06:57:42 +0300
Subject: [PATCH] [SW-86869] gaudi3: add slave fake fuse fail indication

Currently, in case of fake fuse, slave die waits for master
die to complete fake fuse init and then read required values
from scratchpad register. However in case of timeout waiting
for master die to complete fake fuse init, slave die just return
and fuse values for slave die remains 0. This patch adds timeout
indication in ROM status udpate.

Change-Id: I470fa17b85a7cb3e6fa4685838ece7f96ac0d592
---

diff --git a/arch/gaudi3/arc/progress_bits.h b/arch/gaudi3/arc/progress_bits.h
index e7159b5..89e2dd5 100644
--- a/arch/gaudi3/arc/progress_bits.h
+++ b/arch/gaudi3/arc/progress_bits.h
@@ -22,124 +22,125 @@
 	PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL = 15,
 	PROGRESS_INIT_SPI_BOOT_DESC_FAIL = 16,
 	PROGRESS_INIT_EFUSE_SPI = 17,
-	PROGRESS_REVOKE_LOC = 18,
-	PROGRESS_REVOKE_LOC_0 = 19,
-	PROGRESS_REVOKE_LOC_1 = 20,
-	PROGRESS_KEY_LOC = 21,
-	PROGRESS_KEY_LOC_0 = 22,
-	PROGRESS_KEY_LOC_1 = 23,
-	PROGRESS_VER_LOC = 24,
-	PROGRESS_VER_LOC_0 = 25,
-	PROGRESS_VER_LOC_1 = 26,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_START = 27,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_VERSION_FAIL = 28,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_REVOKED = 29,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 30,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_AUTH_FAIL = 31,
-	PROGRESS_SPI_BOOT_DESC_VERIFY_SUCCESS = 32,
-	PROGRESS_BRINGUP_PCI_PRI = 33,
-	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 34,
-	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 35,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 36,
-	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE = 37,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 38,
-	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 39,
-	PROGRESS_BRINGUP_PCI_PRI_DMA = 40,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 41,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 42,
-	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 43,
-	PROGRESS_BRINGUP_PCI_SEC = 44,
-	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 45,
-	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 46,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 47,
-	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 48,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 49,
-	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 50,
-	PROGRESS_BRINGUP_PCI_SEC_DMA = 51,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 52,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 53,
-	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 54,
-	PROGRESS_ARCBOOT_IN_RECOVERY = 55,
-	PROGRESS_ARCBOOT_ROM_TPM = 56,
-	PROGRESS_ARCBOOT_ROM_TPM_INIT_FAIL = 57,
-	PROGRESS_ARCBOOT_ROM_TPM_STARTUP_FAIL = 58,
-	PROGRESS_ARCBOOT_ROM_TPM_PCR_EXTEND_FAIL = 59,
-	PROGRESS_ARCBOOT_ROM_TPM_FINISH = 60,
-	PROGRESS_BRINGUP_TPM_SPI = 61,
-	PROGRESS_BRINGUP_TPM_SPI_PRI = 62,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_TOO_BIG = 63,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_CRC_FAIL = 64,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_DEST_ADDR = 65,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_INVALID_IMG_SIZE = 66,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_IMG_SIZE = 67,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA_ERROR = 68,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA = 69,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_REVOKED = 70,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_AUTH_FAIL = 71,
-	PROGRESS_BRINGUP_TPM_SPI_PRI_SECURITY = 72,
-	PROGRESS_BRINGUP_TPM_SPI_SEC = 73,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_TOO_BIG = 74,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_CRC_FAIL = 75,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_DEST_ADDR = 76,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_INVALID_IMG_SIZE = 77,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_IMG_SIZE = 78,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA_ERROR = 79,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA = 80,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_REVOKED = 81,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_AUTH_FAIL = 82,
-	PROGRESS_BRINGUP_TPM_SPI_SEC_SECURITY = 83,
-	PROGRESS_ARCBOOT_TPM_SPI_FINISH = 84,
-	PROGRESS_ARCBOOT_TPM_SPI_FAIL = 85,
-	PROGRESS_PPBOOT_START = 86,
-	PROGRESS_PPBOOT_FINISH = 87,
-	PROGRESS_BRINGUP_RAM_PRI = 88,
-	PROGRESS_BRINGUP_RAM_PRI_DMA = 89,
-	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 90,
-	PROGRESS_BRINGUP_RAM_SEC = 91,
-	PROGRESS_BRINGUP_RAM_SEC_DMA = 92,
-	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 93,
-	PROGRESS_ARCBOOT_RAM_START = 94,
-	PROGRESS_ARCBOOT_RECOVERY = 95,
-	PROGRESS_ARCBOOT_RECOVERY_DMA = 96,
-	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 97,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 98,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 99,
-	PROGRESS_ARCBOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 100,
-	PROGRESS_ARCBOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 101,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 102,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 103,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 104,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 105,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 106,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 107,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 108,
-	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 109,
-	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 110,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 111,
-	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 112,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 113,
-	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_SIZE = 114,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 115,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 116,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 117,
-	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 118,
-	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 119,
-	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 120,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 121,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 122,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 123,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 124,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 125,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 126,
-	PROGRESS_ARCBOOT_EROM_IMG0 = 127,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 128,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 129,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 130,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_INVALID_IMG_SIZE = 131,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 132,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 133,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 134,
-	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 135,
-	PROGRESS_ARCBOOT_EROM_IMG0_START = 136,
-	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 137,
+	PROGRESS_INIT_EFUSE_SPI_SLV_TIMEDOUT = 18,
+	PROGRESS_REVOKE_LOC = 19,
+	PROGRESS_REVOKE_LOC_0 = 20,
+	PROGRESS_REVOKE_LOC_1 = 21,
+	PROGRESS_KEY_LOC = 22,
+	PROGRESS_KEY_LOC_0 = 23,
+	PROGRESS_KEY_LOC_1 = 24,
+	PROGRESS_VER_LOC = 25,
+	PROGRESS_VER_LOC_0 = 26,
+	PROGRESS_VER_LOC_1 = 27,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_START = 28,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_VERSION_FAIL = 29,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_REVOKED = 30,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 31,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_AUTH_FAIL = 32,
+	PROGRESS_SPI_BOOT_DESC_VERIFY_SUCCESS = 33,
+	PROGRESS_BRINGUP_PCI_PRI = 34,
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 35,
+	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 36,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 37,
+	PROGRESS_BRINGUP_PCI_PRI_INVALID_IMG_SIZE = 38,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 39,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 40,
+	PROGRESS_BRINGUP_PCI_PRI_DMA = 41,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 42,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 43,
+	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 44,
+	PROGRESS_BRINGUP_PCI_SEC = 45,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 46,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 47,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 48,
+	PROGRESS_BRINGUP_PCI_SEC_INVALID_IMG_SIZE = 49,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 50,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 51,
+	PROGRESS_BRINGUP_PCI_SEC_DMA = 52,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 53,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 54,
+	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 55,
+	PROGRESS_ARCBOOT_IN_RECOVERY = 56,
+	PROGRESS_ARCBOOT_ROM_TPM = 57,
+	PROGRESS_ARCBOOT_ROM_TPM_INIT_FAIL = 58,
+	PROGRESS_ARCBOOT_ROM_TPM_STARTUP_FAIL = 59,
+	PROGRESS_ARCBOOT_ROM_TPM_PCR_EXTEND_FAIL = 60,
+	PROGRESS_ARCBOOT_ROM_TPM_FINISH = 61,
+	PROGRESS_BRINGUP_TPM_SPI = 62,
+	PROGRESS_BRINGUP_TPM_SPI_PRI = 63,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_TOO_BIG = 64,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_CRC_FAIL = 65,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_DEST_ADDR = 66,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_INVALID_IMG_SIZE = 67,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_UNALIGNED_IMG_SIZE = 68,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA_ERROR = 69,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_DMA = 70,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_REVOKED = 71,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_KEY_AUTH_FAIL = 72,
+	PROGRESS_BRINGUP_TPM_SPI_PRI_SECURITY = 73,
+	PROGRESS_BRINGUP_TPM_SPI_SEC = 74,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_TOO_BIG = 75,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_CRC_FAIL = 76,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_DEST_ADDR = 77,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_INVALID_IMG_SIZE = 78,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_UNALIGNED_IMG_SIZE = 79,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA_ERROR = 80,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_DMA = 81,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_REVOKED = 82,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_KEY_AUTH_FAIL = 83,
+	PROGRESS_BRINGUP_TPM_SPI_SEC_SECURITY = 84,
+	PROGRESS_ARCBOOT_TPM_SPI_FINISH = 85,
+	PROGRESS_ARCBOOT_TPM_SPI_FAIL = 86,
+	PROGRESS_PPBOOT_START = 87,
+	PROGRESS_PPBOOT_FINISH = 88,
+	PROGRESS_BRINGUP_RAM_PRI = 89,
+	PROGRESS_BRINGUP_RAM_PRI_DMA = 90,
+	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 91,
+	PROGRESS_BRINGUP_RAM_SEC = 92,
+	PROGRESS_BRINGUP_RAM_SEC_DMA = 93,
+	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 94,
+	PROGRESS_ARCBOOT_RAM_START = 95,
+	PROGRESS_ARCBOOT_RECOVERY = 96,
+	PROGRESS_ARCBOOT_RECOVERY_DMA = 97,
+	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 98,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 99,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 100,
+	PROGRESS_ARCBOOT_ROM_PCIE_PHY_SRAM_INIT_TIMEDOUT = 101,
+	PROGRESS_ARCBOOT_ROM_PCIE_PHY_PARA_STS_CLR_TIMEDOUT = 102,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 103,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 104,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 105,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 106,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 107,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 108,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 109,
+	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 110,
+	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 111,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 112,
+	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 113,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 114,
+	PROGRESS_INIT_EROM_BOOT_DESC_INVALID_IMG_SIZE = 115,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 116,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 117,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 118,
+	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 119,
+	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 120,
+	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 121,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 122,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 123,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 124,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 125,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 126,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 127,
+	PROGRESS_ARCBOOT_EROM_IMG0 = 128,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 129,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 130,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 131,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_INVALID_IMG_SIZE = 132,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 133,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 134,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 135,
+	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 136,
+	PROGRESS_ARCBOOT_EROM_IMG0_START = 137,
+	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 138,
 };
diff --git a/arch/gaudi3/arc/slave.c b/arch/gaudi3/arc/slave.c
index 5d3faf9..39dcfeb 100644
--- a/arch/gaudi3/arc/slave.c
+++ b/arch/gaudi3/arc/slave.c
@@ -51,8 +51,10 @@
 			break;
 
 		retry--;
-		if (!retry)
+		if (!retry) {
+			arcboot_rom_status_update(PROGRESS_INIT_EFUSE_SPI_SLV_TIMEDOUT);
 			return;
+		}
 
 		udelay(FUSE_EMULATION_DATA_UDELAY);
 	}
